bvm/ballerina-core/src/main/java/org/ballerinalang/util/transactions/TransactionResourceManager.java

```diff
@@ -158,16 +204,19 @@ public boolean notifyCommit(String transactionId, int transactionBlockId) {
             }
         }
         invokeCommittedFunction(transactionId, transactionBlockId);
-        removeContextsFromRegistry(combinedId);
+        removeContextsFromRegistry(combinedId, transactionId);
+        failedResourceParticipantSet.remove(transactionId);
+        failedLocalParticipantSet.remove(transactionId);
+        localParticipants.remove(transactionId);
         return commitSuccess;
     }
 
     /**
      * This method acts as the callback which aborts all the resources participated in the given transaction.
      *
-     * @param transactionId the global transaction id
+     * @param transactionId      the global transaction id
      * @param transactionBlockId the block id of the transaction
-     * @param isRetryAttempt whether this is a retry attempt
+     * @param isRetryAttempt     whether this is a retry attempt
      * @return the status of the abort operation
      */
     public boolean notifyAbort(String transactionId, int transactionBlockId, boolean isRetryAttempt) {
```

Comment:
Shall we remove this unused parameter isRetryAttempt?

Reply:
I left it because when we clarify the abort/rollback behavior from the coordinator we going to need this.